core/app/models/spree/order.rb

```diff
@@ -424,21 +424,11 @@ def ensure_line_items_are_in_stock
     end
 
     def empty!
-      if completed?
-        raise Spree.t(:cannot_empty_completed_order)
-      else
-        line_items.destroy_all
-        updater.update_item_count
-        adjustments.destroy_all
-        shipments.destroy_all
-        state_changes.destroy_all
-        order_promotions.destroy_all
+      Spree::Dependencies.cart_empty_service.constantize.new.call(order: self)
+    end
 
-        update_totals
-        persist_totals
-        restart_checkout_flow
-        self
-      end
+    def destroy!
+      Spree::Dependencies.cart_destroy_service.constantize.new.call(order: self)
```

Comment:
```suggestion
      Spree::Dependencies.cart_destroy_service.constantize.call(order: self)
```

Reply:
I removed this method because I will be using service in controller (like in #empty)

Reply:
done.